eng/test-configuration.json

```diff
@@ -0,0 +1,5 @@
+{
+  "version": 1,
+  "defaultOnFailure": "rerun",
+  "localRerunCount": 1
```

Comment:
Are the tests in this repo inherently so flaky that rerunning is frequently necessary❔ Suggest at least adding `failureMessage` or `testName` constraints.